### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by nijave** [asottile/dockerfile#20]
Has anyone successfully got this working on Windows? If so, what steps were necessary?

**Comment by asottile** [comment]
It's a little fiddly, but with 32 bit it can work on windows. There's weird things with c go and msvcrt for win64

**Comment by asottile** [comment]
I can provide more concrete steps when I'm not on a phone

**Comment by nijave** [comment]
Trying to build in the Docker Windows build container I get the following

```
PS C:\go\src\github.com\asottile\dockerfile> c:\python\python .\setup.py install
running install
running bdist_egg
running egg_info
creating dockerfile.egg-info
writing dockerfile.egg-info\PKG-INFO
writing dependency_links to dockerfile.egg-info\dependency_links.txt
writing top-level names to dockerfile.egg-info\top_level.txt
writing manifest file 'dockerfile.egg-info\SOURCES.txt'
reading manifest file 'dockerfile.egg-info\SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'dockerfile.egg-info\SOURCES.txt'
installing library code to build\bdist.win-amd64\egg
running install_lib
running build_ext
$ GOPATH='C:\Users\ContainerAdministrator\AppData\Local\Temp\tmprkpwgxfo' go get -d
$ CGO_CFLAGS='-Ic:\python\include -Ic:\python\include' CGO_LDFLAGS='-Lc:\python\libs -lpython36' GOPATH='C:\Users\ContainerAdministrator\AppData\Local\Temp\tmprkpwgxfo' go build -buildmode=c-shared -o 'C:\go\src\github.com\asottile\dockerfile\build\lib.win-amd64-3.6\dockerfile.cp36-win_amd64.pyd'
# _/C_/Users/ContainerAdministrator/AppData/Local/Temp/tmprkpwgxfo/src/github.com/asottile/dockerfile/pylib
In file included from c:\python\include/pytime.h:6:0,
                 from c:\python\include/Python.h:68,
                 from support.c:1:
c:\python\include/object.h:910:17: error: initializer element is not constant
 #define Py_None (&_Py_NoneStruct)
                 ^
support.c:14:34: note: in expansion of macro 'Py_None'
 PyObject* PyDockerfile_Py_None = Py_None;
                                  ^
Traceback (most recent call last):
  File ".\setup.py", line 37, in <module>
    setup_requires=['setuptools-golang>=0.2.0'],
  File "c:\python\lib\site-packages\setuptools\__init__.py", line 143, in setup
    return distutils.core.setup(**attrs)
  File "c:\python\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "c:\python\lib\distutils\dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\install.py", line 67, in run
    self.do_egg_install()
  File "c:\python\lib\site-packages\setuptools\command\install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "c:\python\lib\site-packages\setuptools\command\bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\install_lib.py", line 11, in run
    self.build()
  File "c:\python\lib\distutils\command\install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "c:\python\lib\distutils\cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "c:\python\lib\distutils\dist.py", line 974, in run_command
    cmd_obj.run()
  File "c:\python\lib\site-packages\setuptools\command\build_ext.py", line 78, in run
    _build_ext.run(self)
  File "c:\python\lib\distutils\command\build_ext.py", line 339, in run
    self.build_extensions()
  File "c:\python\lib\distutils\command\build_ext.py", line 448, in build_extensions
    self._build_extensions_serial()
  File "c:\python\lib\distutils\command\build_ext.py", line 473, in _build_extensions_serial
    self.build_extension(ext)
  File "c:\python\lib\site-packages\setuptools_golang.py", line 131, in build_extension
    _check_call(cmd_build, cwd=pkg_path, env=env)
  File "c:\python\lib\site-packages\setuptools_golang.py", line 80, in _check_call
    subprocess.check_call(cmd, cwd=cwd, env=dict(os.environ, **env))
  File "c:\python\lib\subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '('go', 'build', '-buildmode=c-shared', '-o', 'C:\\go\\src\\github.com\\asottile\\dockerfile\\build\\lib.win-amd64-3.6\\dockerfile.cp36-win_amd64.pyd')' returned non-zero exit status 2.
```

**Comment by asottile** [comment]
interesting, can you share your docker setup?

**Comment by nijave** [comment]
I modified the build environment for moby/moby so you can clone that and replace Dockerfile.windows https://gist.github.com/nijave/a4ab48d67468ca1ded5d006ea8d411c5

Then I ran the commands in the other part of the gist (couldn't get git submodules working with the git included in the Docker build so I pull in PortableGit)

**Comment by asottile** [comment]
Thanks! I'll take a look at this :+1: 

I suspect this is a compiler difference between windows and linux that I haven't run into yet -- I should be able to get a fix for that -- especially since it seems to be reproducible.  Thanks for the additional information!

**Comment by asottile** [comment]
should be fixed with 2.0.2! thanks for the issue 🎉
